tests/device_tests/test_sdcard.py

```diff
@@ -43,3 +43,70 @@ def input_flow():
         device.sd_protect(client, Op.ENABLE)
 
     assert e.value.failure.code == messages.FailureType.ProcessError
+
+
+@pytest.mark.sd_card
+@pytest.mark.setup_client(pin="1234")
+def test_sd_protect_unlock(client):
+    def input_flow_enable_sd_protect():
+        yield  # do you really want to enable SD protection
+        assert "SD card protection" in client.debug.wait_layout().text
+        client.debug.press_yes()
+
+        yield  # enter current PIN
+        assert "PinDialog" == client.debug.wait_layout().text
+        client.debug.input("1234")
+
+        yield  # you have successfully enabled SD protection
+        assert "Success" in client.debug.wait_layout().text
+        client.debug.press_yes()
+
+    with client:
+        client.set_input_flow(input_flow_enable_sd_protect)
+        device.sd_protect(client, Op.ENABLE)
+
+    def input_flow_change_pin():
+        yield  # do you really want to change PIN?
+        assert "Change PIN" in client.debug.wait_layout().text
+        client.debug.press_yes()
+
+        yield  # enter current PIN
+        assert "PinDialog" == client.debug.wait_layout().text
+        client.debug.input("1234")
+
+        yield  # enter new PIN
+        assert "PinDialog" == client.debug.wait_layout().text
+        client.debug.input("1234")
+
+        yield  # enter new PIN again
+        assert "PinDialog" == client.debug.wait_layout().text
+        client.debug.input("1234")
+
+        yield  # Pin change successful
+        assert "Success" in client.debug.wait_layout().text
+        client.debug.press_yes()
+
+    with client:
+        client.set_input_flow(input_flow_change_pin)
+        device.change_pin(client)
+
+    client.debug.erase_sd_card(format=False)
+
+    def input_flow_change_pin_format():
+        yield  # do you really want to change PIN?
+        assert "Change PIN" in client.debug.wait_layout().text
+        client.debug.press_yes()
+
+        yield  # SD card problem
+        assert "SD card problem" in client.debug.wait_layout().text
+        client.debug.press_yes()  # retry
```

Comment:
I can't make up my mind, whether this scenario should end with "Wrong SD card" or "SD card problem". I guess both make sense.

Reply:
That's what I think too. I chose this option because it's easier to implement; to report "Wrong SD card" correctly we'd have to be strict in checking that it's the *mounting* that failed, not some other error (failed i/o operation perhaps?)
So i'm going to merge it as-is.
We can make the change in the future if we feel that we should.